idaes/core/initialization/general_hierarchical.py

```diff
@@ -0,0 +1,156 @@
+    """
+
+    CONFIG = InitializerBase.CONFIG()
+    CONFIG.declare(
+        "solver",
+        ConfigValue(
+            default=None,
+            description="Solver to use for initialization",
+        ),
+    )
+    CONFIG.declare(
+        "solver_options",
+        ConfigValue(
+            default={},
+            description="Dict of options to pass to solver",
+        ),
+    )
+
+    def initialization_routine(self, model: Block):
+        """
+        Common initialization routine for models with standard form.
+
+        Args:
+            model: Pyomo Block to be initialized
+
+        Returns:
+            Pyomo solver results object
+        """
+        if not hasattr(model, "control_volume"):
+            raise TypeError(
+                f"Model {model.name} does not appear to be a standard form unit model. "
+                f"Please use an Initializer specific to the model being initialized."
+            )
+
+        # Get logger
+        _log = self.get_logger(model)
+
+        sub_initializers = {}
+        for sm in model._initialization_order:
+            if sm is not model:
+                # Get initializers for add-ons
+                sub_initializers[sm] = self.get_submodel_initializer(sm)
+
+                # Call prepare method for add-ons
+                # TODO: Need arguments for submodel initialization
+                sub_initializers[sm].prepare()
+
+        # Initialize properties
+        try:
+            # Guess a 0-D control volume
+            # Initialize inlet properties - inlet state should already be fixed
+            prop_init = self.get_submodel_initializer(
+                model.control_volume.properties_in
+            )
+
+            if prop_init is not None:
+                prop_init.initialize(
+                    solver=self.config.solver,
+                    optarg=self.config.solver_options,
+                    outlvl=self.config.output_level,
+                )
+
+            # Map solution from inlet properties to outlet properties
+            state = to_json(
+                model.control_volume.properties_in,
+                wts=StoreSpec().value(),
+                return_dict=True,
+            )
+            from_json(
+                model.control_volume.properties_out,
+                sd=state,
+                wts=StoreSpec().value(only_not_fixed=True),
+            )
+        except AttributeError:
+            # Assume it must be a 1-D control volume
+            # TODO: Add steps here
+            raise
+
+        _log.info_high("Step 1: properties initialization complete")
+
+        # Initialize reactions if they exist
+        try:
+            model.control_volume.reactions.initialize(
+                solver=self.config.solver,
+                optarg=self.config.solver_options,
+                outlvl=self.config.output_level,
+                state_vars_fixed=True,
+            )
+        except AttributeError:
```

Comment:
Here and above, it seems a little dangerous to just catch and ignore any attribute error. Ideally, we would have some public API for distinguishing between 0D/1D control volumes and identifying state/reaction blocks.

Reply:
I think this is OK - this is not a 0D/1D issue, but a "does this model have reactions" issue. If there is no block named reactions, then either the model doesn't have any and we can safely `pass`, or the model is not using the standard form(which we have no way of checking) and this initializer is not suitable for it.

Reply:
This also catches and ignores any AttributeError inside the reactions initialize method, which seems dangerous.